src/bin/download_sysext.rs

```diff
@@ -108,6 +112,34 @@ impl<'a> Package<'a> {
             return true;
         }
     }
+
+    fn verify_signature_on_disk(&mut self, from_path: &Path) -> Result<(), Box<dyn Error>> {
+        const PUBKEY_FILE: &str = "../src/testdata/public_key_test_pkcs8.pem";
+
+        let upfile = File::open(from_path)?;
+
+        // Read update payload from file, read delta update header from the payload.
+        let res_data = fs::read_to_string(from_path);
+
+        let header = delta_update::read_delta_update_header(&upfile)?;
+
+        // Extract signature from header.
+        let sigbytes = delta_update::get_signatures_bytes(&upfile, &header)?;
+
+        // Parse signature data from the signature containing data, version, special fields.
+        let _sigdata = match delta_update::parse_signature_data(res_data.unwrap().as_bytes(), &sigbytes, PUBKEY_FILE) {
+            Some(data) => Box::leak(data),
+            _ => {
+                self.status = PackageStatus::BadSignature;
+                return Err("unable to parse signature data".into());
+            }
+        };
+
+        println!("Parsed signature data from file {:?}", from_path);
+
+        self.status = PackageStatus::Verified;
```

Comment:
do we need a call to `verify_sig_pubkey()` first?

Reply:
No, `parse_signature_data()` is already calling `verify_sig_pubkey()`.
https://github.com/flatcar/ue-rs/pull/12/files#diff-4bdabfe18b36b123ce8b3873dae3f39543fc6de6a80b796efddc060d34ef1c60R103